### ISSUE TITLE
🚀 Feature: Change package Dropdown to in-house Dropdown 

### ISSUE THREAD

**Issue by EDWIN1669086** [uclaacm/recursion-lab#238]
### Context
The Dropdown component can be found in this path: src/components/shared/Dropdown.tsx. It has the interface specified below, and uses `Select` from the package `react-select`.

<img width="388" alt="Screenshot 2024-01-23 at 12 43 30 PM" src="https://github.com/uclaacm/recursion-lab/assets/43260874/d7f9789c-ef26-4185-882c-c8e7dbc20612">

### Problem

While this package works. creating our own in-house Dropdown allows us to customize it as we see fit, such as adding local storage features in the future.

### Proposed Solution
Basically, recreate the Dropdown exactly as it is in the package. 

<img width="193" alt="Screenshot 2024-01-23 at 1 07 02 PM" src="https://github.com/uclaacm/recursion-lab/assets/43260874/be43e7e4-b27b-4bf0-9258-4bb6607e35da">

Two changes, though:
- [x] Remove the vertical line separating the text from the dropdown arrow button
- [x] When user clicks dropdown, have the dropdown arrow button flip upward (^)

---

This is a :rocket: Feature Request
